Sources/StreamChat/Models/Attachments/ChatMessageAttachmentSeed.swift

```diff
@@ -55,6 +55,9 @@ public struct ChatMessageAttachmentSeed: Hashable {
         self.type = type
         self.localState = localState
     }
+    
+    // Dummy encodable conformance to satisfy `AttachmentEnvelope` protocol.
+    public func encode(to encoder: Encoder) throws {}
```

Comment:
This is an interesting side effect. I'm thinking - what does this mean? Maybe we're missing something?

Reply:
`Encodable` is the main constraint for the custom attachments cause it is the minimum we need to blindly send it to the backend.
Dummy conformance means that we can't blindly send it, some pre-processing is required and we know how to handle that.
I think that's acceptable trade-off.